File: pid-backend/src/main/java/cz/fit/cvut/pidbackend/Model/Shape.java
package cz.fit.cvut.pidbackend.Model;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "shapes")
@Data
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class Shape implements Serializable {

    @EmbeddedId
    public ShapeId uid;

    @Column(name = "lat")
    private Double lat;
    @Column(name = "lon")
    private Double lon;
    @Column(name = "dist_traveled")
    private Double distTraveled;
}

File: android-app/app/src/main/java/cvut/fit/pidmobapp/model/Shape.kt
package cvut.fit.pidmobapp.model

import androidx.room.PrimaryKey

data class Shape(
    val distTraveled: Double,
    val lat: Double,
    val lon: Double,
    val uid: ShapeUid
) {
    @PrimaryKey(autoGenerate = true)
    var id :Int = 0
}